bok_choy/web_app_test.py

```diff
@@ -34,6 +36,57 @@ def __init__(self, *args, **kwargs):
         self.har_mode = getattr(
             self, 'har_mode', os.environ.get('BOK_CHOY_HAR_MODE', None))
 
+    @classmethod
+    def setUpClass(cls):
+        """
+        Override NeedleTestCase's setUpClass method so that it does not
+        start up the browser once for each testcase class.
+        Instead we start up the browser once per TestCase instance,
+        in the setUp method.
+        """
+        if os.environ.get('NEEDLE_CAPTURE'):
+            cls.capture = True
+        if os.environ.get('NEEDLE_SAVE_BASELINE'):
+            cls.save_baseline = True
+        if os.environ.get('NEEDLE_CLEANUP_ON_SUCCESS'):
+            cls.cleanup_on_success = True
+
+        # Instantiate the diff engine
+        klass = import_from_string(cls.engine_class)
+        cls.engine = klass()
+        super(NeedleTestCase, cls).setUpClass()  # pylint: disable=bad-super-call
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Override NeedleTestCase's tearDownClass method because it
+        would quit the browser. This is not needed as we have already quit the browser
+        after each TestCase, by virtue of a cleanup that we add in the setUp method.
+        """
+        super(NeedleTestCase, cls).tearDownClass()  # pylint: disable=bad-super-call
+
+    def get_web_driver(self):
+        """
+        Override NeedleTestCases's get_web_driver class method to return the WebDriver instance
+        that is already being used, instead of starting up a new one.
+        """
+        return self.browser
+
+    def set_viewport_size(self, width, height):
+        """
+        Override NeedleTestCases's set_viewport_size class method because we need it to operate
```

Comment:
Could you add a bit of doc on what the intention of this is anyway? Not sure what viewport size means.